pkg/logql/engine.go

```diff
@@ -266,6 +266,7 @@ func (q *query) Exec(ctx context.Context) (logqlmodel.Result, error) {
 		Data:       data,
 		Statistics: statResult,
 		Headers:    metadataCtx.Headers(),
+		Warnings:   metadataCtx.Warnings(),
```

Comment:
This is the crux of the PR on the frontend. Everything comes through here so this is where we attach the warnings to be propagated to the scheduler and querier.